telcell/utils/transform.py

```diff
@@ -0,0 +1,222 @@
+    combined_tracks = [(m, 'a') for m in track_a.measurements] + [(m, 'b') for m in track_b.measurements]
+    combined_tracks = sorted(combined_tracks, key=lambda x: x[0].timestamp)
+    paired_measurements = []
+    for (measurement_first, origin_first), (measurement_second, origin_second) in pairwise(combined_tracks):
+        # check this pair is from the two different tracks
+        if origin_first != origin_second:
+            # put the 'a' track first
+            if origin_first == 'a':
+                paired_measurements.append(
+                    MeasurementPair(measurement_first, measurement_second)
+                )
+            elif origin_first == 'b':
+                paired_measurements.append(
+                    MeasurementPair(measurement_second, measurement_first)
+                )
+            else:
+                raise ValueError(f'unclear origin for {origin_first}')
+    return paired_measurements
+
+
+def filter_delay(paired_measurements: List[MeasurementPair],
+                 max_delay: timedelta) \
+        -> List[MeasurementPair]:
+    """
+    Filter the paired measurements based on a specified maximum delay. Can
+    return an empty list.
+
+    :param paired_measurements: A list with all paired measurements.
+    :param max_delay: the maximum amount of delay that is allowed.
+    :return: A filtered list with all paired measurements.
+    """
+    return [x for x in paired_measurements
+            if x.time_difference <= max_delay]
+
+
+def sort_pairs_based_on_rarest_location(
+        switches: List[MeasurementPair],
+        history_track_b: Track,
+        round_lon_lats: bool,
+        max_delay: int = None
+) -> List[Tuple[int, MeasurementPair]]:
+    """
+    Pairs are first filtered on allowed time interval of the two registrations
+    of a single pair. Then, sort pairs based on the rarest location of the
+    track history first and secondly by time difference of the pair.
+
+    :param switches: A list with all paired measurements to consider.
+    :param history_track_b: the history of track_b to find the rarity of locations.
+    :param round_lon_lats: boolean indicating whether to round the lon/lats
+            to two decimals.
+    :param max_delay: maximum allowed time difference (seconds) in a pair.
+                      Default: no max_delay, show all possible pairs.
+    :return: The location counts and measurement pairs that are sorted on the
+            rarest location based on the history and time difference. The
+            location count is the number of occurrences of the coordinates from
+            measurement_b in the track history that is provided.
+
+    TODO There is a problem with testdata, because those are almost continuous
+    lat/lon data, making rarity of locations not as straightforward.
+    Pseudo-solution for now: round lon/lats to two decimals and determine
+    rarity of those.
+    This should not be used if locations are actual cell-ids
```

Comment:
Is er een situatie waarin we op coordinaten willen selecteren? Ik dacht dat we altijd cellid hebben.

Reply:
blijkbaar voor de huidige test data. meer generiek als we niet masten maar eg gps zouden hebben?

Reply:
zo laten voor nu?